api/mapping/services_rules.py

```diff
@@ -841,3 +843,101 @@ def remove_mapping_rules(request, scan_report_id):
     rules = MappingRule.objects.all().filter(scan_report__id=scan_report_id)
 
     rules.delete()
+
+
+def analyse_concepts(scan_report_id):
```

Comment:
It seems like this code is doing a lot and some exception or error might occur (hence the `try`/`except` block). If the unit test set-up is in this branch, could we have a test/tests to show it gets the expected data and that it throws the expected exception/error when error condition is met. If that makes sense?